store/commitment/store.go

```diff
@@ -26,15 +32,16 @@ var (
 // RootStore use a CommitStore as an abstraction to handle multiple store keys
 // and trees.
 type CommitStore struct {
-	logger log.Logger
-
+	logger     log.Logger
+	db         dbm.DB
```

Comment:
why are we using this dependency in store/v2? i thought we would drop it?

Reply:
it is used for commit info metadata storing, if we drop dbm.db, shall we implement store own?

Reply:
We can instead define a `RawDatabase` in `store/database.go` instead of using `dbm.DB`.

Reply:
ok, my concern is we are using `dbm.NewMemdb()` for test purpose, we also should drop this usecase?

Reply:
honestly there's no need. For testing purposes we can just use a very simple memDB

Reply:
@tac0turtle , I have no idea of what's going on `cosmos-db`, could you share more context of it?

Reply:
ideally we archive that repo in the near future. it was a stop gap till this refactor. memdb there evolved from testing to something in prod, we should create a test only memdb here